### PR TITLE
Refactor main.py and add tests

### PR DESCRIPTION
**Pull request by hakancelikdev**

### CODE REVIEW

File: src/unimport/main.py [hunk 1]
```diff
@@ -1,90 +1,146 @@
...
-    )
+    def remove(self, result: _Result, refactor_result):
+        commands.remove(
+            result.path,
+            result.encoding,
+            result.newline,
+            refactor_result,
+            self.config.use_color,
+        )
+        if self.refactor_applied is False:
```
Comment (furkanonder) [inline src/unimport/main.py:103]: There is no need to if statement. You can use ```self.refactor_applied = True```

File: src/unimport/main.py [hunk 2]
```diff
@@ -1,90 +1,146 @@
...
+                color.paint(str(exc), color.RED, self.config.use_color)
                 + " at "
-                + color.paint(path.as_posix(), color.GREEN, config.use_color)
+                + color.paint(
+                    path.as_posix(), color.GREEN, self.config.use_color
+                )
             )
-            is_syntax_error = True
-            continue
+            if self.is_syntax_error is False:
```
Comment (furkanonder) [inline src/unimport/main.py:69]: There is no need to if statement. You can use ```self.is_syntax_error = True```

File: src/unimport/main.py (hunk 1 above)
Comment (hakancelikdev) [reply src/unimport/main.py]: Fix it

File: src/unimport/main.py (hunk 2 above)
Comment (hakancelikdev) [reply src/unimport/main.py]: fix it

**Review by furkanonder — APPROVED** [review]

**Pull request merged**